app/replica.go

```diff
@@ -178,7 +178,11 @@ func startReplica(c *cli.Context) error {
 	dir := c.Args()[0]
 	replicaType := c.String("type")
 	s := replica.NewServer(dir, 512, replicaType)
-	go replica.CreateHoles()
+	go replica.CreateHoles(s)
+	// close replica gracefully upon recieving SIGTERM or SIGINT
+	addShutdown(func() {
```

Comment:
when this func is called on signal, does all other go routines are exited (or) will they be still running?
If they will be still running, those go routines might be doing IOs on the fd available with them.

Reply:
This is handler function, so it will run s.Close upon SIGINT or SIGTERM signals. If any goroutine is already processing s.Close() will be blocked till those operations get finished and then files will be closed gracefully.